### PR TITLE
oc-calendar: Improve check for appointment updates

### PR DESCRIPTION
**Pull request by icandigitbaby**
The UID was being used to check if  the changes in an appointment had been made by
its organiser. In this case, the UID is the user name, without taking the domain into account.
The `owner` variable, however, is a full email address, so the comparison was never successful. This
caused the update notification mail not to be sent.

Suggested `NEWS` line:
`Appointment update notification mails are now sent`

### CODE REVIEW

**Comment by papajulio** [comment]
ok to test

**Comment by papajulio** [comment]
@icandigitbaby I already added you to the list of allowed PR sorry about this

**Comment by blaxter** [comment]
Checking the code I was worried about the case where `newEvent` is nil (in that case it gets the `uid` from `oldEvent`) but I couldn't trigger that behaviour so :+1:

File: SoObjects/Appointments/SOGoAppointmentObject.m [hunk 1]
```diff
@@ -2106,32 +2106,14 @@ - (NSException *) updateContentWithCalendar: (iCalCalendar *) calendar
...
-          
-          if (newEvent == oldEvent)
-            newEvent = nil;
-          
-          if (uid && [uid caseInsensitiveCompare: owner] == NSOrderedSame)
+
+          if ([newEvent userIsOrganizer: ownerUser])
             {
               // A RECCURENCE-ID was removed
               if (!newEvent && oldEvent)
```
Comment (blaxter) [inline SoObjects/Appointments/SOGoAppointmentObject.m]: This cannot happen anymore with this changes

**Comment by icandigitbaby** [comment]
Good catch with the `nil`, I hadn't thought of deletion. Although I haven't been able to reproduce this case, either, I think that the check shouldn't hurt.

**Comment by blaxter** [comment]
:+1:

**Comment by sixstone-qq** [comment]
:+1: and it works.

**Pull request merged**